### PR TITLE
2.4.0 release notes

### PR DESCRIPTION
**Pull request by joejulian**
**What type of PR is this?**

Documentation

**What this PR does/ why we need it**:

Release notes for 2.4.0

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:
image

**Does this PR introduce a user-facing change?**:

```release-note
none
```

**Checklist**

* [x] The commit message explains the changes and why are needed.
* [ ] The code builds and passes lint/style checks locally.
* [ ] The relevant subset of integration tests pass locally.
* [ ] The core changes are covered by tests.
* [x] The documentation is updated where needed.

### CODE REVIEW

File: RELEASE_NOTES.md [hunk 1]
```diff
@@ -1,8 +1,52 @@
 # Release Notes
 
+## stable-1.15-2.4.0, stable-1.16-2.4.0, stable-1.17-2.4.0
+
+* Istio:
```
Comment (samvantran) [inline RELEASE_NOTES.md:5]: Should mention that istio got upgraded to 1.6.8 as that is why all of these changes got pulled in

**Review by samvantran — COMMENTED** [review]
There were a lot of automated chart bumps included in 2.4.0 https://github.com/mesosphere/kubernetes-base-addons/pulls?q=is%3Apr+is%3Amerged+automated

Shouldn't we add some release notes for them?

File: RELEASE_NOTES.md (hunk 1 above)
Comment (shaneutt) [reply RELEASE_NOTES.md:5]: It should be at 1.6.9 for this release now, yes?

**Comment by joejulian** [comment]
> There were a lot of automated chart bumps included in 2.4.0 https://github.com/mesosphere/kubernetes-base-addons/pulls?q=is%3Apr+merged%3A2020-08-26..2020-09-09
> 
> Shouldn't we add some release notes for them?

That's why I tagged this as blocked.

File: RELEASE_NOTES.md (hunk 1 above)
Comment (joejulian) [reply RELEASE_NOTES.md:5]: It's what's in the release notes. /shrug

**Comment by joejulian** [comment]
> There were a lot of automated chart bumps included in 2.4.0 https://github.com/mesosphere/kubernetes-base-addons/pulls?q=is%3Apr+merged%3A2020-08-26..2020-09-09
> 
> Shouldn't we add some release notes for them?

Went through every one of them. Only elasticsearch had release notes and was broken. The rest were `none`.

File: RELEASE_NOTES.md [hunk 2]
```diff
@@ -1,8 +1,63 @@
 # Release Notes
 
+## stable-1.15-2.4.0, stable-1.16-2.4.0, stable-1.17-2.4.0
+
+* - Elasticsearch:
+    - Fixes regression from [helm/charts&#35;17643](https://github.com/helm/charts/pull/17643) where the explicit selectors do match the previously implicit selectors.
+    - Fix plugin install initcontainer which would fail if plugin already exists. This happens when Node reboots and keeps emptyDir, or if elasticsearch image already contains plugin
+    - In private environments where we replicate all the images, the test image cannot be pulled due it misses imagePullSecrets ([#497](https://github.com/mesosphere/kubernetes-base-addons/pull/497), [@d2iq-dispatch](https://github.com/d2iq-dispatch))
```
Comment (samvantran) [inline RELEASE_NOTES.md:8]: Didn't we revert this bc its broken? https://github.com/mesosphere/kubernetes-base-addons/pull/512

If the revert isn't included in the release, shouldn't it be? Its going to break upgrades.

**Review by samvantran — CHANGES_REQUESTED** [review]
I dont think we want the elastic change in or we missed including the revert.

**Review by samvantran — APPROVED** [review]

**Review by shaneutt — APPROVED** [review]

**Pull request merged**